File: src/main/java/app/pdfx/FieldID.java
package app.pdfx;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)

public @interface FieldID {
    String value();

    enum FieldType {
        STRING,
        TEXT,
        LONG,
        INT,
        DATE,
        BOOL
    }

    FieldType type() default FieldType.STRING;
}

File: src/main/kotlin/app/pdfx/FieldId.kt
package app.pdfx

@Retention(AnnotationRetention.RUNTIME)
annotation class FieldID(val value: String, val type: FieldType = FieldType.STRING) {
    enum class FieldType {
        STRING,
        TEXT,
        LONG,
        INT,
        DATE,
        BOOL
    }
}
